source/grammar/index.rst

```diff
@@ -80,53 +88,59 @@ repetition is denoted by curly brackets ``+`` (one or more) and ``*`` (zero or m
     incrementor: "++" | "--
     pragma: "#pragma {" ... "}"
     declaration_statement: (quantum_declaration | classical_declaration | constant_declaration) ";"
-    assignment_expression: expression assignment_operator expression
-    binary_operator: "+" | "-" | "*" | "/" 
+    assignment_expression: "=" expression
+    binary_operator: "+" | "-" | "*" | "/"
         :| "<<" | ">>" | "rotr" | "rotl" | "&" | "|" | "^" | "&&" | "||"
         :| ">" | "<" | ">=" | "<=" | "==" | "!="
     assignment_operator: "=" | "+=" | "-=" | "*=" | "/=" |
         :| "<<=" | ">>=" |  "&=" | "|=" | "^=" | "~=" | "->"
     membership_test: id "in" set_declaration
-    index_set_declaration: "{" index_set_generators "}"
-    index_set_generators: expression_list | range
-    range: expression? ":" expression? ( ":" expression )?
+    index_set_declaration: "{" expression_list "}" | range
+    range: "[" expression? ":" expression? ( ":" expression )? "]"
     selection_statement: "if (" expression ")" program_block ( "else" program_block )?
-    iteration_statement: "for" membership_test program_block | "while (" expression ")" program_block
+    iteration_statement: ( "for" membership_test | "while (" expression ")" ) program_block
     selection_directive_statement: selection_directive ";"
     selection_directive: "break" | "continue" | "exit"
     quantum_gate_definition: "gate" quantum_gate_signature quantum_gate_block
-    quantum_gate_signature: id quantum_gate_args id_lis
-    quantum_gate_args: ( "(" classical_type_and_id_list? ")" )?
+    quantum_gate_signature: id classical_params id_list
+    id_list: ( id "," )* id
+    index_type: id ( "[" expression "]" )?
```

Comment:
Should this just be a `classical_type_declaration`? and the actual arguments a list of these?

Reply:
Which piece are you talking about? Are you talking about `classical_params`?

If so, I use an optional list of `classical_type_and_id_list` because it has the `association` component`.

Reply:
Fixed in https://github.com/Qiskit/openqasm/pull/89/files.